lib/routes/account.js

```diff
@@ -973,78 +977,89 @@ module.exports = (log, db, mailer, Password, config, customs, signinUtils, push)
         }
         request.setMetricsFlowCompleteSignal(flowCompleteSignal)
 
-        return resetAccountData()
+        return checkRecoveryKey()
+          .then(resetAccountData)
+          .then(deleteRecoveryKey)
           .then(createSessionToken)
           .then(createKeyFetchToken)
           .then(recordSecurityEvent)
           .then(createResponse)
           .then(reply, reply)
 
+        function checkRecoveryKey() {
+          if (recoveryKeyId && wrapKb) {
```

Comment:
The `&& wrapKb` is redundant here, we know from route validation that it will always be true when `recoveryKeyId` is true.